### PR TITLE
Add Vision API connection.

### PR DESCRIPTION
**Pull request by daspecster**
As per a conversation about small PR's with @dhermes, here is Vision's connection class.

This is pretty simple, just adding the needed scope.

### CODE REVIEW

**Comment by dhermes** [comment]
Looks fine but Travis is failing.

File: gcloud/vision/connection.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+                         :attr:`Connection.API_BASE_URL`.
+    """
+
+    API_BASE_URL = 'https://vision.googleapis.com'
+    """The base of the API call URL."""
+
+    API_VERSION = 'v1'
+    """The version of the API, used in building the API call's URL."""
+
+    API_URL_TEMPLATE = '{api_base_url}/{api_version}{path}'
```
Comment (dhermes) [inline gcloud/vision/connection.py]: Can you give an example URL?

**Comment by daspecster** [comment]
It was failing, I believe because I didn't have the RST for it in there. Also I added an API URL example. `https://vision.googleapis.com/v1/images:annotate`

File: scripts/generate_json_docs.py [hunk 2]
```diff
@@ -629,7 +629,8 @@ def main():
             'resource_manager': [],
             'storage': [],
             'streaming': [],
-            'translate': []
+            'translate': [],
+            'vision': []
```
Comment (dhermes) [inline scripts/generate_json_docs.py]: Can you add a trailing comma

File: gcloud/vision/connection.py (hunk 1 above)
Comment (dhermes) [inline gcloud/vision/connection.py]: FWIW I just meant give me an example so I can verify the template. i.e. not needed in the docstring

**Comment by dhermes** [comment]
LGTM once Travis passes though you should squash if not too inconvenient

File: gcloud/vision/connection.py (hunk 1 above)
Comment (daspecster) [inline gcloud/vision/connection.py]: Oh ok, so you just wanted https://vision.googleapis.com/v1/images:annotate?

I can take it back out if you want.

**Comment by daspecster** [comment]
Wow travis...what's the deal?

**Comment by tseaver** [comment]
@daspecster I'm guessing the conflict is in `docs/json/json/master/toc.json`, after @dhermes merged #2171.

**Comment by dhermes** [comment]
Yeah Travis has been slow. And in the 35 minutes since you posted that, I did give you a conflict. Sorry.

**Comment by daspecster** [comment]
Conflict resolved....back to waiting on travis.

**Comment by daspecster** [comment]
Finally!

**Pull request merged**